kotlin-inject-compiler/core/src/main/kotlin/me/tatarka/inject/compiler/TypeCollector.kt

```diff
@@ -280,7 +280,9 @@ class TypeCollector(private val provider: AstProvider, private val options: Opti
             val methods = mutableListOf<AstMember>()
             val isProvides = mutableMapOf<AstMember, Boolean>()
             for (method in allMethods) {
-                val existing = methods.firstOrNull { it.name == method.name && it.signatureEquals(method) }
+                val existing = methods.firstOrNull {
+                    it.name == method.name && (it.signatureEquals(method) || it.overrides(method))
```

Comment:
I was thinking of doing something with `KsTypeReference.resolve.isAssignableFrom` but that didn't fit in `signatureEquals` (since it's technically not equal) and `overrides` was the previous behavior.